src/Inventory/Asset/MainAsset.php

```diff
@@ -925,6 +925,11 @@ public function handleAssets()
             foreach ($assets as $asset) {
                 $asset->setEntityID($this->getEntityID());
                 $asset->setExtraData($this->assets);
+                foreach ($this->assets as $asset_type => $asset_list) {
+                    if ($asset_type != '\\' . get_class($asset)) {
```

Comment:
Maybe checking with `is_a` would permit to use both `OperatingSystem::class` and `'\Glpi\Inventory\Asset\OperatingSystem'` values, and make code more secure.
```suggestion
                    if (!is_a($asset, $asset_type)) {
```

Reply:
I guess this is more complex and would require code change at other places. This is an issue since the beginning